File: wimg-ios/wimg/Models/Category.swift
import SwiftUI

enum WimgCategory: Int, CaseIterable, Identifiable {
    case uncategorized = 0
    case groceries = 1
    case dining = 2
    case transport = 3
    case housing = 4
    case utilities = 5
    case entertainment = 6
    case shopping = 7
    case health = 8
    case insurance = 9
    case income = 10
    case transfer = 11
    case cash = 12
    case subscriptions = 13
    case travel = 14
    case education = 15
    case other = 255

    var id: Int { rawValue }

    var name: String {
        switch self {
        case .uncategorized: "Unkategorisiert"
        case .groceries: "Lebensmittel"
        case .dining: "Essen gehen"
        case .transport: "Transport"
        case .housing: "Wohnen"
        case .utilities: "Nebenkosten"
        case .entertainment: "Unterhaltung"
        case .shopping: "Shopping"
        case .health: "Gesundheit"
        case .insurance: "Versicherung"
        case .income: "Einkommen"
        case .transfer: "Überweisung"
        case .cash: "Bargeld"
        case .subscriptions: "Abos"
        case .travel: "Reisen"
        case .education: "Bildung"
        case .other: "Sonstiges"
        }
    }

    var color: Color {
        switch self {
        case .uncategorized: Color(hex: 0xDFE6E9)
        case .groceries: Color(hex: 0x4ECDC4)
        case .dining: Color(hex: 0xFF6B6B)
        case .transport: Color(hex: 0x45B7D1)
        case .housing: Color(hex: 0x96CEB4)
        case .utilities: Color(hex: 0xA8D8EA)
        case .entertainment: Color(hex: 0xDDA0DD)
        case .shopping: Color(hex: 0xF7DC6F)
        case .health: Color(hex: 0xFF9FF3)
        case .insurance: Color(hex: 0xC8D6E5)
        case .income: Color(hex: 0x2DC653)
        case .transfer: Color(hex: 0xB8B8B8)
        case .cash: Color(hex: 0xFFD93D)
        case .subscriptions: Color(hex: 0x6C5CE7)
        case .travel: Color(hex: 0xFD79A8)
        case .education: Color(hex: 0x74B9FF)
        case .other: Color(hex: 0xDFE6E9)
        }
    }

    var icon: String {
        switch self {
        case .uncategorized: "questionmark.circle"
        case .groceries: "cart"
        case .dining: "fork.knife"
        case .transport: "car"
        case .housing: "house"
        case .utilities: "bolt"
        case .entertainment: "tv"
        case .shopping: "bag"
        case .health: "heart"
        case .insurance: "shield"
        case .income: "arrow.down.circle"
        case .transfer: "arrow.left.arrow.right"
        case .cash: "banknote"
        case .subscriptions: "repeat"
        case .travel: "airplane"
        case .education: "book"
        case .other: "ellipsis.circle"
        }
    }

    static func from(_ value: Int) -> WimgCategory {
        WimgCategory(rawValue: value) ?? .uncategorized
    }
}

extension Color {
    init(hex: UInt32) {
        self.init(
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0
        )
    }
}

File: wimg-android/app/src/main/java/com/wimg/app/models/Category.kt
package com.wimg.app.models

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.*
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector

enum class WimgCategory(
    val id: Int,
    val label: String,
    val color: Color,
    val icon: ImageVector,
) {
    UNCATEGORIZED(0, "Sonstiges", Color(0xFF8E8E93), Icons.Outlined.HelpCenter),
    GROCERIES(1, "Lebensmittel", Color(0xFF34C759), Icons.Outlined.ShoppingCart),
    DINING(2, "Essen gehen", Color(0xFFFF9500), Icons.Outlined.Restaurant),
    TRANSPORT(3, "Transport", Color(0xFF007AFF), Icons.Outlined.DirectionsCar),
    HOUSING(4, "Wohnen", Color(0xFFAF52DE), Icons.Outlined.Home),
    UTILITIES(5, "Nebenkosten", Color(0xFF5AC8FA), Icons.Outlined.Bolt),
    ENTERTAINMENT(6, "Unterhaltung", Color(0xFFFF2D55), Icons.Outlined.SportsEsports),
    SHOPPING(7, "Shopping", Color(0xFFFF6482), Icons.Outlined.ShoppingBag),
    HEALTH(8, "Gesundheit", Color(0xFFFF3B30), Icons.Outlined.FavoriteBorder),
    INSURANCE(9, "Versicherung", Color(0xFF64D2FF), Icons.Outlined.Shield),
    INCOME(10, "Einkommen", Color(0xFF30D158), Icons.Outlined.NorthEast),
    TRANSFER(11, "Überweisung", Color(0xFF8E8E93), Icons.Outlined.SwapHoriz),
    CASH(12, "Bargeld", Color(0xFFFFD60A), Icons.Outlined.Payments),
    SUBSCRIPTIONS(13, "Abonnements", Color(0xFFBF5AF2), Icons.Outlined.Subscriptions),
    TRAVEL(14, "Reisen", Color(0xFF30B0C7), Icons.Outlined.Flight),
    EDUCATION(15, "Bildung", Color(0xFF5856D6), Icons.Outlined.School),
    OTHER(255, "Andere", Color(0xFF8E8E93), Icons.Outlined.MoreHoriz);

    companion object {
        fun fromId(id: Int): WimgCategory = entries.find { it.id == id } ?: UNCATEGORIZED
    }
}
